### PR TITLE
INS-3761: fix traceID propagation

### PR DESCRIPTION
**Pull request by sexyhamster**
**- What I did**
fixed trace propagation for custom traces

### CODE REVIEW

File: instrumentation/instracer/tracer.go [hunk 1]
```diff
@@ -98,10 +98,10 @@ func StartSpanWithSpanID(ctx context.Context, name string, spanID uint64, o ...o
 			traceID = sc.TraceID()
 			parentID = sc.SpanID()
 		}
-	} else if traceStr := inslogger.TraceID(ctx); traceStr != "" {
+	} else if traceStr := inslogger.TraceID(ctx); len(traceStr) >= 16 {
```
Comment (ivanshibitov) [inline instrumentation/instracer/tracer.go]: may move this magic number to constant?

File: virtual/integration/main_test.go [hunk 2]
```diff
@@ -128,7 +128,7 @@ func NewServer(
 	receiveCallback func(meta payload.Meta, pl payload.Payload) []payload.Payload,
 	mManager machinesmanager.MachinesManager) (*Server, error) {
 
-	traceID := "main_" + utils.RandTraceID()
+	traceID := utils.RandTraceID() + "_main"
```
Comment (ivanshibitov) [inline virtual/integration/main_test.go]: What for?

File: instrumentation/instracer/tracer.go [hunk 3]
```diff
@@ -169,17 +169,16 @@ func ParentSpanCtx(ctx context.Context) (jaeger.SpanContext, context.Context) {
...
-		traceID, err = jaeger.TraceIDFromString(string(traceSpan.TraceID))
-		if err != nil {
-			inslogger.FromContext(ctx).Error(errors.Wrap(err, "failed to parse tracespan traceID"))
-			return emptyContext, ctx
+		stringTrace = string(traceSpan.TraceID)
+	}
+
+	if len(stringTrace) >= 16 {
+		if len(stringTrace) > 32 {
+			stringTrace = stringTrace[:31]
```
Comment (ivanshibitov) [inline instrumentation/instracer/tracer.go]: magic constants everywhere

File: network/utils.go [hunk 4]
```diff
@@ -208,7 +208,7 @@ func IsClosedPipe(err error) bool {
 }
 
 func NewPulseContext(ctx context.Context, pulseNumber uint32) context.Context {
-	insTraceID := "pulse_" + strconv.FormatUint(uint64(pulseNumber), 10)
+	insTraceID := strconv.FormatUint(uint64(pulseNumber), 10) + "_pulse"
```
Comment (ivanshibitov) [inline network/utils.go]: поч?

File: instrumentation/instracer/tracer.go (hunk 1 above)
Comment (sexyhamster) [reply instrumentation/instracer/tracer.go]: these constants are hardcoded inside jaeger tracer. So there is no point of having them around

File: virtual/integration/main_test.go (hunk 2 above)
Comment (sexyhamster) [reply virtual/integration/main_test.go]: to be able to track this trace with jaeger (I cut everything after 32 symbols). But still have them verbose in logs

File: network/utils.go (hunk 4 above)
Comment (sexyhamster) [reply network/utils.go]: for consitency, switced from prefixes to suffixes

File: instrumentation/instracer/tracer.go (hunk 3 above)
Comment (sexyhamster) [reply instrumentation/instracer/tracer.go]: moved to constants and added comment for >= case

File: instrumentation/instracer/tracer.go [hunk 5]
```diff
@@ -169,17 +175,17 @@ func ParentSpanCtx(ctx context.Context) (jaeger.SpanContext, context.Context) {
...
-			traceSpan.TraceID = traceSpan.TraceID[:32]
-		}
-		traceID, err = jaeger.TraceIDFromString(string(traceSpan.TraceID))
-		if err != nil {
-			inslogger.FromContext(ctx).Error(errors.Wrap(err, "failed to parse tracespan traceID"))
-			return emptyContext, ctx
+		stringTrace = string(traceSpan.TraceID)
+	}
+
+	// If it is shorter then uint32, then probably it is custom trace and it cannot be used for jaeger trace
```
Comment (ivanshibitov) [inline instrumentation/instracer/tracer.go]: :heart:

**Pull request merged**